src/Block/ProfileMenuBlockService.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Block;
+
+use Knp\Menu\ItemInterface;
+use Knp\Menu\Provider\MenuProviderInterface;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\MenuBlockService;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Hugo Briand <briand@ekino.com>
+ */
+final class ProfileMenuBlockService extends MenuBlockService
+{
+    private $menuBuilder;
+
+    /**
+     * @param object $menuBuilder
+     */
+    public function __construct(string $name, EngineInterface $templating, MenuProviderInterface $menuProvider, $menuBuilder)
+    {
+        parent::__construct($name, $templating, $menuProvider, []);
+
+        if (!\is_object($menuBuilder) || !\is_callable([$menuBuilder, 'createProfileMenu'])) {
+            throw new \InvalidArgumentException(
+                'Argument 4 should be object with public function "createProfileMenu(array $itemOptions = [])"'
+            );
+        }
+
+        $this->menuBuilder = $menuBuilder;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
Please also remove the docblock
```suggestion
    public function getName(): string
```

Reply:
https://github.com/sonata-project/SonataUserBundle/pull/1136#discussion_r364696008 It is correct.